07_errors/sirigithub/dbstore.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+type dbStore struct {
+	dbfilePath string
+	currID     int
+}
+
+func NewdbStore(filePath string) *dbStore {
+	var store = dbStore{}
+	store.dbfilePath = filePath
+	return &store
```

Comment:
You can implement this function with a single statement:
```
return &dbStore{dbfilePath: filePath}
```

However, it would probably be better to open the database when the dbStore is created, and add a `Close() error` method so the caller can close it later (which implements the `io.Closer` interface:
```
type dbStore struct {
  conn *db.leveldb.DB
  currID int
}

func NewdbStore(filePath string) *dbStore {
    return &dbStore{conn: dbConn(filePath)} // one day I'll handle errors here :-)
}

func (s *dbStore) Close() error {
    return s.conn.Close()
}
```

But you'll also need a different approach for `currID`. Because the database is persistent, if the process restarts, you start allocating IDs from 1 again. These will conflict with existing IDs.

A different approach is to use `github.com/google/uuid` to generate random IDs. That way do you not need to maintain any state to allocate IDs.